genesyscloud/resource_genesyscloud_telephony_providers_edges_phonebasesettings.go

```diff
@@ -0,0 +1,261 @@
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"properties": {
+				Description:      "phone base settings properties",
+				Type:             schema.TypeString,
+				Optional:         true,
+				Computed:         true,
+				DiffSuppressFunc: suppressEquivalentJsonDiffs,
+			},
+			"capabilities": {
+				Description: "Phone Capabilities.",
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
+				Computed:    true,
+				Elem:        phoneCapabilities,
+			},
+			"line_base_settings_id": {
+				Description: "Computed line base settings id",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+		},
+		CustomizeDiff: customizePhoneBaseSettingsPropertiesDiff,
+	}
+}
+
+func createPhoneBaseSettings(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+	phoneMetaBase := buildSdkDomainEntityRef(d, "phone_meta_base_id")
+	properties := buildBaseSettingsProperties(d)
+
+	phoneBase := platformclientv2.Phonebase{
+		Name:          &name,
+		PhoneMetaBase: phoneMetaBase,
+		Properties:    properties,
+		Lines: &[]platformclientv2.Linebase{
+			{
+				Name:         &name,
+				LineMetaBase: phoneMetaBase,
+			},
+		},
+		Capabilities: buildSdkCapabilities(d),
+	}
+
+	if description != "" {
+		phoneBase.Description = &description
+	}
+
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	edgesAPI := platformclientv2.NewTelephonyProvidersEdgeApiWithConfig(sdkConfig)
+
+	log.Printf("Creating phone base settings %s", name)
+	phoneBaseSettings, _, err := edgesAPI.PostTelephonyProvidersEdgesPhonebasesettings(phoneBase)
+	if err != nil {
+		return diag.Errorf("Failed to create phone base settings %s: %s", name, err)
+	}
+
+	d.SetId(*phoneBaseSettings.Id)
+
+	log.Printf("Created phone base settings %s", *phoneBaseSettings.Id)
+
+	return readPhoneBaseSettings(ctx, d, meta)
+}
+
+func updatePhoneBaseSettings(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+	phoneMetaBase := buildSdkDomainEntityRef(d, "phone_meta_base_id")
+	properties := buildBaseSettingsProperties(d)
+	id := d.Id()
+
+	phoneBase := platformclientv2.Phonebase{
+		Id:            &id,
+		Name:          &name,
+		PhoneMetaBase: phoneMetaBase,
+		Properties:    properties,
+		Lines: &[]platformclientv2.Linebase{
+			{
+				Name:         &name,
+				LineMetaBase: phoneMetaBase,
+			},
+		},
+		Capabilities: buildSdkCapabilities(d),
+	}
+
+	if description != "" {
+		phoneBase.Description = &description
+	}
+
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	edgesAPI := platformclientv2.NewTelephonyProvidersEdgeApiWithConfig(sdkConfig)
+
+	phoneBaseSettings, resp, getErr := edgesAPI.GetTelephonyProvidersEdgesPhonebasesetting(d.Id())
+	if getErr != nil {
+		if resp != nil && resp.StatusCode == 404 {
```

Comment:
Go's http package has a set of constants for this.  I personally would use this rather then a 404.  Small thing.

Reply:
Yeah fair point. I'll make a ticket to have magic numbers for HTTP codes and API calls removed